src/usr/local/sbin/openvpn.learn-address.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+# openvpn learn-address script maintaining DNS entries of connected clients in
+# unbound config.
+
+DOMAIN="$1"
+OP="$2"
+IP="$3"
+CN="$4"
+
+DIR="/var/unbound"
+CONF="$DIR/openvpn.client.$IP.conf"
+PIDFILE="/var/run/unbound.pid"
+
+case "$OP" in
+
+	add|update)
+		# Remove all configs which mention the FQDN
+		grep -l -null -F "local-data: \"$CN.$DOMAIN A $IP\"" $DIR/openvpn.client.*.conf | xargs -0 rm
```

Comment:
Use braces around shell variables to reduce ambiguity, for example: `${CN}.${DOMAIN}` and so on. Many instances of this throughout the script.